### PR TITLE
[Typing][B-20] Add type annotations for `python/paddle/distribution/lognormal.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
B20

### CODE REVIEW

File: python/paddle/distribution/lognormal.py [hunk 1]
```diff
@@ -89,14 +108,16 @@ class LogNormal(TransformedDistribution):
                 [0.34939718])
     """
 
-    def __init__(self, loc, scale):
+    def __init__(
+        self, loc: _LognormalBoundary, scale: _LognormalBoundary
+    ) -> None:
```
Comment (megemini) [inline python/paddle/distribution/lognormal.py]: 是不是缺了类属性 `loc scale`？

Comment (enkilee) [reply python/paddle/distribution/lognormal.py]: 抱歉，已加

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Review by SigureMo — CHANGES_REQUESTED** [review]
等待 https://github.com/PaddlePaddle/Paddle/pull/65849#discussion_r1671718328 确认

**Review by SigureMo — DISMISSED** [review]

**Review by SigureMo — APPROVED** [review]

**Pull request merged**